### PR TITLE
 fix: RTL unicode issue in PDF

### LINKED ISSUE #873: RTL breakage in PDF
### Describe the issue

Archived-At: <https://mailarchive.ietf.org/arch/msg/auth48archive/_8ZjoHcPh5HfYxRQb3P4zVbdXUg>

  https://www.rfc-editor.org/authors/rfc9290.txt
  https://www.rfc-editor.org/authors/rfc9290.pdf
  https://www.rfc-editor.org/authors/rfc9290.html
  https://www.rfc-editor.org/authors/rfc9290.xml

This is likely to lead to TI state.

### Code of Conduct

- [X] I agree to follow the [IETF's Code of Conduct](https://github.com/ietf-tools/.github/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by kesara**
Fixes #873

### CODE REVIEW

File: xml2rfc/writers/pdf.py [hunk 1]
```diff
@@ -61,6 +62,14 @@ def __init__(self, xmlrfc, quiet=None, options=default_options, date=None):
         else:
             ftlogger.setLevel(logging.ERROR)
 
+    def flatten_unicode_spans(self, html):
+        # This is a fix for bug in WeasyPrint that doesn't handle RTL unicode
+        # content correctly.
+        # See #873 & Kozea/WeasyPrint#1711
+        return re.sub(r'<span class="unicode">(?P<unicode_content>.*)</span>',
```
Comment (cabo) [inline xml2rfc/writers/pdf.py:69]: Shouldn't this be a `.*?` so it doesn't bleed over into additional spans?
Maybe add a test case with `<u>a</u>b<u>c</u>` as well.

Comment (kesara) [reply xml2rfc/writers/pdf.py:69]: Yes, you are correct. It's greedy by default. I'll fix that and add an unit test.

Comment (kesara) [reply xml2rfc/writers/pdf.py]: I've updated the commit.

Comment (cabo) [reply xml2rfc/writers/pdf.py]: LGTM now.

**Review by rjsparks — APPROVED** [review]

**Pull request merged**